subprojects/dependency-management/src/main/java/org/gradle/api/internal/attributes/DefaultAttributesSchema.java

```diff
@@ -141,6 +143,22 @@ public void addConsumerDescriber(AttributeDescriber describer) {
         consumerAttributeDescribers.add(describer);
     }
 
+    @Override
+    public void attributeDisambiguationPrecedence(Attribute<?>... attributes) {
+        precedence.addAll(Arrays.asList(attributes));
+    }
+
+    @Override
+    public void setAttributeDisambiguationPrecedence(Attribute<?>... attributes) {
+        precedence.clear();
+        precedence.addAll(Arrays.asList(attributes));
```

Comment:
Consider invoking `attributeDisambiguationPrecedence(attributes)` instead? This limits the number of places we mutate `precedence`.